src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -548,22 +548,17 @@ public ImmutableArray<T> InsertRange(int index, IEnumerable<T> items)
             {
                 return ImmutableArray.CreateRange(items);
             }
-
-            int count = ImmutableExtensions.GetCount(ref items);
-            if (count == 0)
+            
+            T[] other = items.ToArray();
```

Comment:
:exclamation: I'm not sure this is better. It only iterates once, but it allocates a new intermediate array. When `items` implements `ICollection<T>`, I don't see how this is necessarily better.